### PR TITLE
MAGECLOUD-3248: Add control over docker mounts

### PR DESCRIPTION
**Pull request by sikharam-adb**
### Description

Add control over Docker mounts.

### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-3248

### Manual testing scenarios

1. Clone a cloud template.
2. Make the following changes to the `composer.json` file.
```
    "repositories": {
        "repo": {
            "type": "composer",
            "url": "https://repo.magento.com"
        },
        "ece-tools": {
            "type": "vcs",
            "url": "https://github.com/magento/ece-tools"
        },
        "docker": {
            "type": "vcs",
            "url": "https://github.com/magento/magento-cloud-docker"
        }
    },
    "require": {
        "magento/magento-cloud-metapackage": ">=2.3.2 <2.3.3",
        "magento/ece-tools": "dev-develop as 2002.0.99",
        "magento/magento-cloud-docker": "dev-MAGECLOUD-3248 as 1.0.99"
    }
```
3. Run `composer update`.
4. Generate the docker configuration file `docker-compose.yml` with `./vendor/bin/ece-docker build:compose`.
5. Launch docker with `docker-compose up -d`.
6. Build and deploy Magento in the container with `docker-compose run build cloud-build` and `docker-compose run deploy cloud-deploy` respectively.
7. Access local Magento (https://magento2.docker/) instance without errors.
8. Inspect the web container `docker inspect <project>_web_1` (for example) and confirm that the `mounts` defined in the `.magento.app.yaml` of the project are present in the container `Mounts`.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

**Comment by BaDos** [comment]
It looks like good.

I just have a proposal to rename readers:
```
/**
 * @var Reader
 */
private $reader;

/**
 * @var ConfigReader
 */
private $configReader;
```

These names can confuse developers.

Maybe will be better to change namespaces for these readers:
`Magento\CloudDocker\Config\Environment\Reader` -> `Magento\CloudDocker\Config\Magento\SharedReader`

`Magento\CloudDocker\Config\Reader` -> `Magento\CloudDocker\Config\Application\Reader`

Or something like this.

@shiftedreality @oshmyheliuk @NadiyaS @sikharam-adb @billygilbert @bbatsche @michaelcasey316 What do you think?

File: src/Test/Unit/Config/Environment/ReaderTest.php [hunk 1]
```diff
@@ -8,7 +8,7 @@
 namespace Magento\CloudDocker\Test\Unit\Config\Environment;
```
Comment (BaDos) [inline src/Test/Unit/Config/Environment/ReaderTest.php]: Is this test for `Magento\CloudDocker\Config\Environment\Shared\Reader`?
If yes - need to change the namespace from `Magento\CloudDocker\Test\Unit\Config\Environmen` to `Magento\CloudDocker\Test\Unit\Config\Environmen\Shared`

File: src/Test/Unit/Config/ReaderTest.php [hunk 2]
```diff
@@ -8,7 +8,7 @@
 namespace Magento\CloudDocker\Test\Unit\Config;
```
Comment (BaDos) [inline src/Test/Unit/Config/ReaderTest.php]: The same, need to change the namespace.

**Review by BaDos — APPROVED** [review]

**Review by shiftedreality — APPROVED** [review]

**Comment by mveeramneni** [comment]
QA Approved

**Pull request merged**